vscode/src/chat/chat-view/handlers/ChatHandler.ts

```diff
@@ -348,3 +359,74 @@ function combineContext(
 ): ContextItem[] {
     return [explicitMentions, openCtxContext, priorityContext, retrievedContext].flat()
 }
+
+/**
+ * promptCachingProcessing combines codebase context messages and add prompt caching to the codebase context.
+ * We combine the codebase context to only use 1 cache breakpoint in total (limit is 4 cache breakpoints).
+ * Docs: https://docs.anthropic.com/en/docs/build-with-claude/prompt-caching
+ */
+function promptCachingProcessing(messages: Message[]): Message[] {
+    const messageMap: Record<string, Message> = {}
+    const roleFirstIndex: Record<string, number> = {}
+    const finalMessages: Message[] = []
+
+    let count = 0
+    let i = 0
+    let combinedContext = false
+
+    while (i < messages.length) {
```

Comment:
```suggestion
    for (let i = 0; i < messages.length; i++) {
```

Would it be easier if we do for loop hereif we do for loop here so we don't have to manually keep track of the index?

Reply:
Done